receiver/oracledbreceiver/factory.go

```diff
@@ -65,6 +67,16 @@ func createReceiverFunc(sqlOpenerFunc sqlOpenerFunc, clientProviderFunc clientPr
 	}
 }
 
+func getDataSource(cfg Config) string {
+	if cfg.DataSource != "" {
+		return cfg.DataSource
+	}
+
+	// Data source string is format of:
+	// oracle://username:password@endpoint/OracleDBService
+	return fmt.Sprintf("oracle://%s:%s@%s/%s", cfg.Username, cfg.Password, cfg.Endpoint, cfg.Service)
```

Comment:
Given that we parse the password using url.Parse, it would be great to check that we can use a password with special characters (@ and : for example). I think we need to encode the password before inserting it into the string.

Reply:
If allowed, it should probably go through https://pkg.go.dev/net/url#PathEscape

Reply:
I'll update the password option to allow for special characters.

From a UX perspective: Would it be preferable to allow users to encode special characters in the `datasource` option as well? So we'd only be calling `url.Parse` after we call `url.PathUnescape` when parsing the `datasource`?

Reply:
> From a UX perspective: Would it be preferable to allow users to encode special characters in the datasource option as well? So we'd only be calling url.Parse after we call url.PathUnescape when parsing the datasource?

Not sure about this. I'm good with any approach as long as it's documented

Reply:
I would lead with a few tests that check that passwords with special characters are supported, that's what I have had as feedback as a pain point.

Reply:
I've added a note in the README, special characters must be encoded in the `datasource` string. I've also added tests to ensure special characters work in the password, and are properly encoded into the datasource string when connecting to the Oracle DB instance.